### PR TITLE
MBL-81: set backend up to deliver socket updates in correct format 

### PR DESCRIPTION
**Pull request by richchurcher**
This:
  - [x] returns a slightly modified shape from `updateUserSocketRoom` which removes the guesswork I used when I wrote it the first time around
  - [x] returns a much more detailed post object from `createPost`
  - [x] migrates `link_preview_id` from `integer` to `bigint`
  - [x] adds some helpers to `api/models/util` to simplify some of the boilerplate

### CODE REVIEW

**Comment by levity** [comment]
won't this affect code in evo that's expecting the current shape?

**Comment by levity** [comment]
never mind -- i just tried it in evo and the changes are minor enough that they don't disrupt any existing code

**Comment by richchurcher** [comment]
@levity Yeah the changes are modelled off the graphql request we make in evo, which I used essentially unchanged for mobile... so I think we're ok. Plus I don't think socket updates are wired in yet for evo notifications.

I do have one possible addition/change for the `newNotificationsCount` so I'll hold off merging for a bit.

File: api/models/Notification.js [hunk 1]
```diff
@@ -298,19 +298,30 @@ module.exports = bookshelf.Model.extend({
...
       this.actor().pick([ 'id', 'name' ]),
       { avatarUrl: this.actor().get('avatar_url') }
     )
+    const meta = activity.get('meta')
+    const action = Notification.priorityReason(meta.reasons)
     const comment = this.comment().pick([ 'id', 'text' ])
-    const community = this.relations.activity.relations.community.pick([ 'id', 'name', 'slug' ])
+    const community = activity.relations.community.pick([ 'id', 'name', 'slug' ])
+    const createdAt = activity.get('created_at').toString()
     const post = { id: this.post().id, title: this.post().get('name') }
```
Comment (levity) [inline api/models/Notification.js]: minor quibble -- if this line is only used in the final payload object, i'd rather just see it listed in-line there. but up to you

**Comment by richchurcher** [comment]
@levity time for another look? Obviously needs tests.

File: api/models/Post.js [hunk 2]
```diff
@@ -259,8 +260,40 @@ module.exports = bookshelf.Model.extend(Object.assign({
...
+        commentsTotal: 0,
+        communities: refineMany(communities, [ 'id', 'name', 'slug' ]),
+        creator,
+        linkPreview: refineOne(linkPreview, [ 'id', 'image_url', 'title', 'url' ]),
+        topics,
+
+        // May need to retain these to avoid breaking legacy code.
+        // TODO: Check if these are still required.
+        creatorId: creator.id,
+        tags: topics
```
Comment (levity) [inline api/models/Post.js]: Yep, these two are still required in SocketListener on evo: https://github.com/Hylozoic/hylo-evo/blob/master/src/components/SocketListener/SocketListener.store.js#L55

**Pull request merged**